openedx/core/djangoapps/credit/partition_schemes.py

```diff
@@ -0,0 +1,120 @@
+"""
+Provides partition support to the user service.
+"""
+
+import logging
+
+from course_modes.models import CourseMode
+from student.models import CourseEnrollment
+from lms.djangoapps.verify_student.models import SkippedReverification, VerificationStatus
+
+
+log = logging.getLogger(__name__)
+
+
+class VerificationPartitionScheme(object):
+    """
+    This scheme randomly assigns users into the partition's groups.
+    """
+    NON_VERIFIED = 'non_verified'
+    VERIFIED_ALLOW = 'verified_allow'
+    VERIFIED_DENY = 'verified_deny'
+
+    @classmethod
+    def get_group_for_user(cls, course_key, user, user_partition):
+        """
+        Return the user's group depending their enrollment and verification
+        status.
+
+        Args:
+            user(User): user object
+            course_id(CourseKey): CourseKey
+
+        Returns:
+            Boolean
+        """
+        checkpoint = user_partition.parameters["location"]
+
+        if (
+            not _is_enrolled_in_verified_mode(user, course_key) or
+            _was_denied_at_any_checkpoint(user, course_key)
+        ):
+            return cls.NON_VERIFIED
+        elif (
+            _has_skipped_any_checkpoint(user, course_key) or
+            _has_completed_checkpoint(user, course_key, checkpoint)
+        ):
+            return cls.VERIFIED_ALLOW
+        else:
+            return cls.VERIFIED_DENY
+
+    @classmethod
+    def key_for_partition(cls, xblock_location_id):
+        """
+        Returns the key to use to look up and save the user's group for a given user partition.
+        """
+        return 'verification:{0}'.format(xblock_location_id)
+
+
+def _is_enrolled_in_verified_mode(user, course_key):
```

Comment:
Since these methods are no longer private methods of class so you can remove `_` from start of the names of these methods. That way you will not need to suppress pylint warning for private member access.